### PR TITLE
stat/distmv: rename NewNormalChol and add Sigma as an input argument

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
I'm not sure this is the best name, but I can't think of anything better.

**Comment by btracey** [comment]
Fixes #47

File: stat/distmv/normal.go [hunk 1]
```diff
@@ -64,19 +64,25 @@ func NewNormal(mu []float64, sigma mat.Symmetric, src *rand.Rand) (*Normal, bool
 	return n, true
 }
 
-// NewNormalChol creates a new Normal distribution with the given mean and
-// covariance matrix represented by its Cholesky decomposition. NewNormalChol
-// panics if len(mu) is not equal to chol.Size().
-func NewNormalChol(mu []float64, chol *mat.Cholesky, src *rand.Rand) *Normal {
+// NewNormalData creates a new Normal distribution with the given mean and
+// covariance matrix when the Cholesky decomposition of the covariance matrix
+// is already known. NewNormalChol panics if len(mu) is not equal to chol.Size()
```
Comment (kortschak) [inline stat/distmv/normal.go]: s/NewNormalChol/NewNormalData/

Also in the test name.

**Comment by btracey** [comment]
Do you have any ideas for the name?

**Comment by kortschak** [comment]
No unfortunately - I would have suggested one if I did.

**Comment by kortschak** [comment]
What exactly is the reason for retaining this now since #47 says it's signficant raison d'être of saving allocation no longer exists?

**Comment by btracey** [comment]
It saves the cost of computing Sigma, which is an O(n^3). Thinking though, it may be better to export particular functions (like LogProb), where the Cholesky is already known. This saves all of the overhead (allocation, computation of sigma, etc.)

**Comment by btracey** [comment]
We've done more of the latter. If there's a compelling use case, we can reconsider.

**Pull request closed without merging**